### ISSUE TITLE
Remove old static Window Functions page

### ISSUE THREAD

**Issue by alamb** [apache/datafusion#13171]
Part of https://github.com/apache/datafusion/issues/12740

### Is your feature request related to a problem or challenge?

@Omega359  added the infrastructure to automatically create documentation from code in https://github.com/apache/datafusion/pull/12668

After https://github.com/apache/datafusion/pull/13161 Window Functions still have two pages

1. A "static" one: https://datafusion.apache.org/user-guide/sql/window_functions.html
2. A "auto generated" one https://datafusion.apache.org/user-guide/sql/window_functions_new.html

### Describe the solution you'd like

Once all the window functions have all been migrated
- https://github.com/apache/datafusion/issues/8709

We should remove the static window functions page (in favor of the auto generated one). We can follow the model of https://github.com/apache/datafusion/pull/13161

### Describe alternatives you've considered

Basically this means for the scalar, window and aggregate functions:
1. Delete the static file `docs/source/user-guide/sql/window_functions.md`
2. Rename the `_new` files like `docs/source/user-guide/sql/window_functions_new.md` to  `docs/source/user-guide/sql/window_functions.md`
3. Update the index page `docs/source/user-guide/sql/index.rst` to remove the `_new` entries
4. Update the script `dev/update_function_docs.sh` so it writes to the scalar_functions.md file

**Comment by jonathanc-n** [comment]
This is just waiting on #12649

**Comment by buraksenn** [comment]
It seems assignee removed their assignment from that issue. Let me implement that and remove old static windows then

**Comment by alamb** [comment]
@jonathanc-n  notes that now we have merged the following PR this is unblocked 🐱 🚀 
- https://github.com/apache/datafusion/pull/13201
